pallets/teebag/src/lib.rs

```diff
@@ -0,0 +1,833 @@
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			AllowSGXDebugMode::<T>::put(self.allow_sgx_debug_mode);
+			Mode::<T>::put(self.mode);
+			if let Some(ref admin) = self.admin {
+				Admin::<T>::put(admin);
+			}
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		// Needed for the conversion of `mrenclave` to a `Hash`.
+		// The condition holds for all known chains.
+		<T as frame_system::Config>::Hash: From<[u8; 32]>,
+	{
+		/// Set the admin account
+		///
+		/// Weights should be 2 DB writes: 1 for mode and 1 for event
+		#[pallet::call_index(0)]
+		#[pallet::weight((2 * T::DbWeight::get().write, DispatchClass::Normal, Pays::No))]
+		pub fn set_admin(
+			origin: OriginFor<T>,
+			new_admin: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			Admin::<T>::put(new_admin.clone());
+			Self::deposit_event(Event::AdminSet { new_admin: Some(new_admin) });
+			Ok(Pays::No.into())
+		}
+
+		/// Set the mode
+		///
+		/// Weights should be 2 DB writes: 1 for mode and 1 for event
+		#[pallet::call_index(1)]
+		#[pallet::weight((2 * T::DbWeight::get().write, DispatchClass::Normal, Pays::Yes))]
+		pub fn set_mode(
+			origin: OriginFor<T>,
+			new_mode: OperationalMode,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			Mode::<T>::put(new_mode);
+			Self::deposit_event(Event::ModeSet { new_mode });
+			Ok(().into())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn set_max_enclave_count(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+			new_count: u64,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(
+				new_count > Self::max_enclave_count(worker_type),
+				Error::<T>::MaxEnclaveCountUnderflow
+			);
+
+			MaxEnclaveCount::<T>::insert(worker_type, new_count);
+			Self::deposit_event(Event::MaxEnclaveCountSet { worker_type, new_count });
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_add_enclave(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			enclave: Enclave,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			Self::add_enclave(&who, &enclave)?;
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_remove_enclave(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			Self::remove_enclave(&who)?;
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_remove_enclave_by_mrenclave(
+			origin: OriginFor<T>,
+			mrenclave: MrEnclave,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
```

Comment:
I assume they can register again, is it ok ?

Reply:
They can - do you want to blacklist them?

If so we could adjust the `ScheduledEnclave` entry and/or the `MaxEnclaveCount`

Reply:
No, just thinking about the purpose of these calls. 
If the purpose is just sweeping the dust, then the mess can easily come back again - even with blacklisting I think.